src/api/entities/SecurityToken/Checkpoints/__tests__/index.ts

```diff
@@ -87,30 +81,66 @@ describe('Checkpoints class', () => {
       sinon.restore();
     });
 
-    test('should return all created checkpoints with their timestamps', async () => {
-      const timestamps = [1000, 2000, new Date().getTime() + 10000];
-      const ids = [1, 2, 3];
-      const rawTicker = dsMockUtils.createMockTicker(ticker);
-
-      dsMockUtils.createQueryStub('checkpoint', 'timestamps', {
-        entries: timestamps.map((timestamp, index) =>
-          tuple(
-            [rawTicker, dsMockUtils.createMockU64(ids[index])],
-            dsMockUtils.createMockMoment(timestamp)
-          )
-        ),
-      });
+    test('should return all created checkpoints with their timestamps and total supply', async () => {
+      const stringToTickerStub = sinon.stub(utilsConversionModule, 'stringToTicker');
+
+      dsMockUtils.createQueryStub('checkpoint', 'totalSupply');
+
+      const requestPaginatedStub = sinon.stub(utilsInternalModule, 'requestPaginated');
+
+      const totalSupply = [
+        {
+          ticker: 'FAKETICKER',
+          checkpointId: new BigNumber(1),
+          balance: new BigNumber(100),
+          moment: new Date('10/10/2020'),
+        },
+        {
+          ticker: 'OTHERTICKER',
+          checkpointId: new BigNumber(2),
+          balance: new BigNumber(1000),
+          moment: new Date('11/11/2020'),
+        },
+      ];
+
+      totalSupply.forEach(({ ticker: t }) =>
+        stringToTickerStub.withArgs(t, context).returns(dsMockUtils.createMockTicker(t))
+      );
```

Comment:
The ticker would always be the same since you're getting the checkpoints for a specific token. I see no need in having it defined for each entry